lib/msf/core/post/file.rb

```diff
@@ -398,6 +398,29 @@ def upload_file(remote, local)
     write_file(remote, ::File.read(local))
   end
 
+  #
+  # Upload a binary and write it as +remote+ on the remote file
+  # system
+  #
+  # @param remote [String] Destination file name on the remote filesystem
+  # @param data [String] Data to be uploaded
+  # @return (see #write_file)
+  def upload_binary(path, data)
```

Comment:
What does this method offer over using `write_file` ?

Reply:
It has a print_status. The idea was to refactor in some of the existing upload_binary methods that do the same thing (after we land this).